### PR TITLE
test/l4lb,nat46x64: Replace Kind/Helm with DinD

### PR DESCRIPTION
**Pull request by brb**
See commit msgs.

Successful run - https://github.com/cilium/cilium/actions/runs/3685163577/jobs/6235811777.

### CODE REVIEW

**Review by borkmann — APPROVED** [review]
💯

**Comment by brb** [comment]
The nat46x64 test case is failing after the rewrite :-( 

The following curl fails - https://github.com/cilium/cilium/pull/22653/files#diff-a971eb8f1b155111301db7bc65b14d8f369bc570b09cec47f67ff24cbd54e914R125:
```
+[11:37:33] curl -o /dev/null 10.0.0.4:80
```
The `cilium monitor` output (started just before the curl):
```
<- network flow 0x75fdd831 , identity unknown->unknown state unknown ifindex eth0 orig-ip 0.0.0.0: 172.12.42.1:47222 -> 10.0.0.4:80 tcp SYN

xx drop (FIB lookup failed, 7) flow 0x0 to endpoint 0, ifindex 13, file 105:833, , identity unknown->unknown: [2001:db8:1::2]:47222 -> [2001:db8:1::3]:80 tcp SYN
```

Err=7 == `BPF_FIB_LKUP_RET_NO_NEIGH`.

The PERM neigh entry was added before:
```
+[11:36:52] nsenter -t 5725 -n ip neigh add 172.12.42.3 dev eth0 lladdr 02:42:ac:0c:2a:03
+[11:36:52] nsenter -t 5725 -n ip neigh add 2001:db8:1::3 dev eth0 lladdr 02:42:ac:0c:2a:03
```

@borkmann Any idea?

**Comment by borkmann** [comment]
[...]
> ```
> xx drop (FIB lookup failed, 7) flow 0x0 to endpoint 0, ifindex 13, file 105:833, , identity unknown->unknown: [2001:db8:1::2]:47222 -> [2001:db8:1::3]:80 tcp SYN
> ```
> 
> Err=7 == `BPF_FIB_LKUP_RET_NO_NEIGH`.
> 
> The PERM neigh entry was added before:
> 
> ```
> +[11:36:52] nsenter -t 5725 -n ip neigh add 172.12.42.3 dev eth0 lladdr 02:42:ac:0c:2a:03
> +[11:36:52] nsenter -t 5725 -n ip neigh add 2001:db8:1::3 dev eth0 lladdr 02:42:ac:0c:2a:03
> ```
> 
> @borkmann Any idea?

Could you dump the next hop address that fib lookup returned? Maybe its different than expected.

**Review by nbusseneau — APPROVED** [review]
LGTM. I like this change, clarifies the purpose of the tests.

File: test/l4lb/test.sh [hunk 1]
```diff
@@ -30,58 +28,52 @@ clang -O2 -Wall -target bpf -c bpf_xdp_veth_host.c -o bpf_xdp_veth_host.o
...
 
-# Extract the internal URL to the apiserver from the kind config
-kind get kubeconfig --internal > internal-kubeconfig
-HOSTPORT=$(kubectl --kubeconfig internal-kubeconfig config view -o jsonpath='{.clusters[0].cluster.server}')
-HOSTPORT=$(echo ${HOSTPORT} | cut -d/ -f3)
-HOST=$(echo ${HOSTPORT}| cut -d: -f1)
-PORT=$(echo ${HOSTPORT} | cut -d: -f2)
-
+# Wait until Docker is ready in the lb-node node
+while ! docker exec -t lb-node docker ps >/dev/null; do sleep 1; done
```
Comment (aspsk) [inline test/l4lb/test.sh:54]: Does this test have external timeout? If not, you can do something like
```
TIMEOUT=60 # 1 minute or so
for dummy in $(seq $TIMEOUT); do
    docker exec -t lb-node docker ps >/dev/null && break
done
if [ "$dummy" = "$TIMEOUT" ]; then
    blah-blah-blah
    exit 1
fi
```

File: test/l4lb/test.sh [hunk 2]
```diff
@@ -145,4 +145,6 @@ for i in $(seq 1 10); do
 done
 
 # Cleanup
-kind delete cluster
+docker rm -f lb-node
+docker rm -f nginx
+docker network rm cilium-l4lb
```
Comment (aspsk) [inline test/l4lb/test.sh:150]: May this test break any consequent test if we don't clean the docker setup? If yes, then we better do a `trap EXIT` to cleanup

**Review by aspsk — APPROVED** [review]
Left two questions, but more for my curiosity

**Review by tklauser — APPROVED** [review]
:rocket:

**Pull request merged**